lib/tasks/import_images.rake

```diff
@@ -0,0 +1,10 @@
+namespace :import_images do
+  desc "Imports placeholder images using the Faker gem"
+  task generate_profile_pictures: :environment do
+    # Randomly grab half of all Authors, and set their profile images to Faker data;
+    Author.all.sample(Author.all.length/2).each do |author| 
```

Comment:
Prefer using `Author.count` to get the count of records, `#count` will use a DB query whereas `#length` will load all the records into an array in memory, and then count the size of the array.

Also, something to be aware of is that `sample` as used here is also an array method, so it will _also_ load all the records into memory. If you have a large `authors` table this could be prohibitively expensive, in terms of the DB query itself, the RAM needed to fit all those ActiveRecord objects in memory, and the time needed to create them.

Unfortunately, selecting a truly random subset of data from a database isn't something that has a definite "best practice" solution (here's an example stack-overflow of just how complicated solutions can get: https://stackoverflow.com/questions/8674718/best-way-to-select-random-rows-postgresql). That said, for this specific exercise, there's no requirement that a truly _random_ subset of authors get profile images... setting profile images for the first half is equally valid, and can be done in a much more performant way.

Last thought, when looping over any arbitrarily large set of records with ActiveRecord, prefer using `find_each`, which loads a batch of records into memory, processes them, then loads the next batch. This helps keep memory usage and query time down.